src/Module/RenderCartBookingInfoHandler.php

```diff
@@ -189,19 +193,37 @@ protected function _renderBookingInfo(BookingInterface $booking)
     {
         $format = $this->_containerGet($this->cartItemConfig, 'booking_datetime_format');
 
+        try {
+            $clientTz = $this->_containerGet($this->_normalizeContainer($booking), 'client_tz');
+        } catch (InvalidArgumentException $invalidArgumentException) {
+            $clientTz = 'UTC';
+        } catch (NotFoundExceptionInterface $notFoundException) {
+            $clientTz = 'UTC';
+        }
+
+        try {
+            $tz = new DateTimeZone($clientTz);
```

Comment:
Should this default to UTC, or would it be better to let the error bubble up?

Reply:
UTC, or even server timezone, would be a better option IMO. The timezone is retrieved from the booking record, but it is not required. So failing doesn't make sense in the absence of the data.

The timezone name may also be invalid, such as an empty string. A browser is not required to expose such information, and in EDD Bookings we retrieve a customer's timezone from the browser. To me this is again not an error, but just a case where the booking has no client timezone attached to it.

Reply:
So, in the event where a valid timezone cannot be acquired, you will display the time in UTC, but you will display the timezone itself too, right?

Reply:
No - just displaying the time in UTC. The timezone is not shown anywhere.

Also, upon further consideration, perhaps server timezone would be more appropriate.

Reply:
I'm not sure I understand why timezone is not required. It appears that if the timezone cannot be acquired, and it is not shown, then this should result in an error.

Reply:
It is not required because it's absence does not break the logic or any other part of a booking system. It is used for display purposes only, and never for any form of booking-related calculations, unlike `start`, `end`, `service_id`, `resource_id` and `status`. 

Given that the method of acquiring it is also not guaranteed to succeed, it logically follows that a fallback value is more ideal than an error, which the user might not now how to remedy.

This has already been agreed upon in the past. So to me now, the only question that needs answering is what fallback value makes most sense, which to me now seems to be the server's timezone to match the times shown on the admin side and potentially also on the front-side (consider EDD Bookings services with the local time display option turned off).

Reply:
I agree that fallback makes sense, as opposed to an error. But I think that showing the timezone also makes a lot of sense - even more so if it is a fallback timezone, i.e. not something that was explicitly set anywhere.

Reply:
Kindly discuss this with Mark. In the meantime, I will implement a decent fallback mechanism with injection. Sounds good?

Reply:
@markzahra, what do you think about showing the offset?

Reply:
We should show the timezone being used in the checkout page.

Reply:
A new line under "Until" will work for now.